ingest/expression_files/expression_files.py

```diff
@@ -109,3 +119,20 @@ def set_data_array_gene_expression_values(
             self.study_file_id,
         ).get_data_array():
             yield model
+
+    def load(self, gene_docs: List, data_array_docs: List):
+        start_time = datetime.datetime.now()
+        self.insert(gene_docs, self.COLLECTION_NAME)
+        self.insert(data_array_docs, 'data_array')
+        print(f'Time to load {len(gene_docs) + len(data_array_docs)} models: {str(datetime.datetime.now() - start_time)}')
+
+    def insert(self, docs: List, collection_name: str):
+        try:
+            self.mongo_connection._client[collection_name].insert_many(
+                docs,  ordered=False)
+        except BulkWriteError as bwe:
+            print(f"Error caused by inserting into collection '{collection_name}': {bwe.details}")
+            raise BulkWriteError(f'Error caused by data docs : {bwe.details}')
+        except Exception as e:
+            print(f"Error caused by inserting into collection '{collection_name}': {e}")
+            raise Exception(f'{e}')
```

Comment:
You should raise the exception with the additional message you have above.  Alternatively, just do `raise e` so that you preserve the type of the root error

Reply:
a45ee4f